src/zenml/zen_stores/rest_zen_store.py

```diff
@@ -3979,76 +3949,165 @@ def update_tag(
     # Internal helper methods
     # =======================
 
-    def _get_auth_token(self) -> str:
-        """Get the authentication token for the REST store.
+    @property
+    def api_token(self) -> str:
```

Comment:
IMO this being a property hides away too much of the complexity of what's happening in this method. This not only does HTTP calls but also writes to the credentials store, while it seems like just a simple attribute access when you for example do
```python
self.session.headers.update(
    {"Authorization": "Bearer " + self.api_token}
)
```

Maybe we can make this a method and call it `get_or_generate_api_token`?

Reply:
Fair point !